overlord/snapstate/snapstate_install_test.go

```diff
@@ -4246,3 +4246,71 @@ func (s *snapmgrTestSuite) TestInstallDeduplicatesSnapNames(c *C) {
 	c.Check(installed, testutil.DeepUnsortedMatches, []string{"some-snap", "some-base"})
 	c.Check(ts, HasLen, 2)
 }
+
+type installFn func(info *snap.SideInfo) (*state.TaskSet, error)
+
+func (s *snapmgrTestSuite) TestCorrectNumRevisionsIfNoneAdded(c *C) {
+	// different paths to install a revision already stored in the state
+	installFuncs := []installFn{
+		func(si *snap.SideInfo) (*state.TaskSet, error) {
+			yaml := "name: some-snap\nversion: 1.0\nepoch: 1*"
+			path := snaptest.MakeTestSnapWithFiles(c, yaml, nil)
+			ts, _, err := snapstate.InstallPath(s.state, si, path, "some-snap", "", snapstate.Flags{})
+			return ts, err
+		}, func(si *snap.SideInfo) (*state.TaskSet, error) {
+			return snapstate.Update(s.state, "some-snap", &snapstate.RevisionOptions{Revision: si.Revision}, s.user.ID, snapstate.Flags{})
+		},
+	}
+
+	for _, fn := range installFuncs {
+		s.testRetainCorrectNumRevisions(c, fn)
+	}
+}
+
+func (s *snapmgrTestSuite) testRetainCorrectNumRevisions(c *C, installFn installFn) {
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	seq := []*snap.SideInfo{{
+		RealName: "some-snap",
+		SnapID:   "some-snap-id",
+		Revision: snap.R(1),
+	}, {
+		RealName: "some-snap",
+		SnapID:   "some-snap-id",
+		Revision: snap.R(2),
+	}}
```

Comment:
I think we do have a test that checks installation of **different** revisions and retaining the correct number of them right, no need to test that?

Reply:
True, this can be simplified to just one revision. Thanks

Reply:
Hmm, that's fine but it wasn't my question :)

Reply:
Oh sorry, I misunderstood then. I thought your point was that the test doesn't need more than one revision, it just needs to test that installing the same revision doesn't the discard the existing one

Reply:
Yes, sorry, indeed my question wasn't clear. I was asking whether we need to check for regular cases of garbage-collection of old revisions. But I just checked and we have a bunch od update tests about that already, so it's fine.